crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -109,6 +110,48 @@ impl SemanticSyntaxChecker {
         }
     }
 
+    fn irrefutable_match_case<Ctx: SemanticSyntaxContext>(stmt: &ast::Stmt, ctx: &Ctx) {
+        let Stmt::Match(ast::StmtMatch { cases, .. }) = stmt else {
+            return;
+        };
```

Comment:
Should we move the match expression in `check_stmt` above now that we have multiple checks? That would also require updating the methods like `irrefutable_match_case` to accept the relevant AST node which needs to be checked like in this case it would accept `StmtMatch` which would also lead to a safer typed API.